javatests/arcs/core/data/expression/ExpressionTest.kt

```diff
@@ -72,13 +72,43 @@ class ExpressionTest {
         assertThat(evalNum(2.toBigInteger().asExpr() * 2.asExpr())).isEqualTo(4.toBigInteger())
         assertThat(evalNum(6.toBigInteger().asExpr() / 3.asExpr())).isEqualTo(2.toBigInteger())
     }
+
     @Test
     fun evaluate_fieldOps() {
         // field ops
         assertThat(evalNum(num("blah"))).isEqualTo(10)
         assertThat(evalNum(scope("baz").get<Number>("x"))).isEqualTo(24)
     }
 
+    @Test
+    fun evaluate_fieldOps_nullSafe() {
+        val paxelExpr = PaxelParser.parse("from f in foos select first(f.words)?.word")
+            as Expression<Sequence<String?>>
+
+        val scope = CurrentScope(
+            mutableMapOf(
+                "foos" to listOf(
+                    mapOf("words" to listOf(
+                        mapOf("word" to "Lorem").asScope(),
+                        mapOf("word" to "ipsum").asScope()
+                    )).asScope(),
+                    mapOf("words" to listOf<Scope>()).asScope(),
```

Comment:
I don't understand how this is workign without patching First() to use firstOrNull() internally. If you call first on an emptySequence() it is supposed to through an exception:

```
/**
 * Returns first element.
 * @throws [NoSuchElementException] if the sequence is empty.
 *
 * The operation is _terminal_.
 */
public fun <T> Sequence<T>.first(): T {
    val iterator = iterator()
    if (!iterator.hasNext())
        throw NoSuchElementException("Sequence is empty.")
    return iterator.next()
}
```

Reply:
#6108 changed first to firstOrNull() merged just an hour ago